### PR TITLE
Support for testing JobStorage, Queuename, and JobConfig

### PR DESCRIPTION
**Pull request by raulriera**
Hi,

I notice this repo while I was searching for background jobs in Vapor, wanted to contribute back by adding a few tests. Maybe some of them can be removed (like the JSON one, but I think it would be good to start nailing all the functions down with their respective tests)

Thanks for the contributions to Vapor ❤️ 

~~Hmmm, unsure how to fix those Linux import issues~~

### CODE REVIEW

File: Tests/JobsTests/JobStorageTests.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+//
+
+import XCTest
+@testable import Jobs
+
+final class JobStorageTests: XCTestCase {
+	func testStringRepresentationIsValidJSON() {
+		let jobStorage = JobStorage(key: "vapor", data: Data(), maxRetryCount: 1, id: "identifier", jobName: "jobs")
+		
+		XCTAssertEqual(jobStorage.stringValue(), "{\"data\":\"\",\"key\":\"vapor\",\"maxRetryCount\":1,\"id\":\"identifier\",\"jobName\":\"jobs\"}")
```
Comment (mcdappdev) [inline Tests/JobsTests/JobStorageTests.swift]: I like the idea behind this test, but the `stringValue()` function is deterministic so we can't rely on the ordering being the same every time. Perhaps we can decode the string value back into a new struct and compare the properties?

**Comment by raulriera** [comment]
Hey! sure thing, just a few questions

> Would you mind running this through Xcode's indentation when you get a chance? It seems like there is some inconsistency with spacing.

It seems that your code was written with a different indentation properties, are you using 2 spaces? I ran the indentation but it made no difference for me (Tabs and 4 spaces) 

> Looks like we're failing on CircleCI because of a rogue import statement. Want to check that out?

In the description I'm actually asking for some help here, I don't know what is causing this issues, and running `swift test` in docker is not returning any errors.

**Comment by mcdappdev** [comment]
Hmmm - this is what I'm seeing on my end: 
Screen Shot 2019-03-17 at 2 09 06 PM

Oops - I missed your point about the imports in the description. What if you try running `swift test --generate-linuxmain`?

**Comment by raulriera** [comment]
Oh wow, I was making those changes by hand :O, didn't know about `generate-linuxmain` thanks! will try that.

Ah, spaces I see... I'll make the changes

File: Tests/JobsTests/JobStorageTests.swift [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        let stringRepresentation = jobStorage.stringValue()
+        
+        if let data = stringRepresentation?.data(using: String.Encoding.utf8), let jobStorageRestored = try? JSONDecoder().decode(JobStorage.self, from: data) {
+            XCTAssertEqual(jobStorage.key, jobStorageRestored.key)
+            XCTAssertEqual(jobStorage.data, jobStorageRestored.data)
+            XCTAssertEqual(jobStorage.maxRetryCount, jobStorageRestored.maxRetryCount)
+            XCTAssertEqual(jobStorage.id, jobStorageRestored.id)
+            XCTAssertEqual(jobStorage.jobName, jobStorageRestored.jobName)
+        } else {
+            XCTFail("There was a problem restoring JobStorage")
```
Comment (raulriera) [inline Tests/JobsTests/JobStorageTests.swift]: Usually I don't like this, open to suggestions

**Comment by raulriera** [comment]
👋 ready for another review

**Comment by penny-coin** [comment]
Hey @raulriera, you just merged a pull request, have a coin! 

You now have 1 coins.

**Pull request merged**